sentry/src/main/java/io/sentry/SentryTransaction.java

```diff
@@ -204,6 +205,16 @@ public void setDescription(@Nullable String description) {
     return this.context;
   }
 
+  @Override
+  public void setData(@NotNull String key, @NotNull Object value) {
```

Comment:
Is it worth adding setData if event has setExtra already?
The other SDKs where transactions is just an event (without the separation we did here) will have just `setExtra` in the docs and samples?

Reply:
https://github.com/getsentry/sentry-cocoa/pull/932#discussion_r573211307

Reply:
Made a comment there. I think adding stuff to transaction such as that field isn't the way to go since this isn't what the protocol has. It'll be confusing to have both Extra and Data.

We have to remember that in reality Transaction is an Event, we just turned things around here to simplify (not sure it did at this point) things.

Reply:
It had to be added here because `ITransaction` extends `ISpan`. Docs also say that each transaction is a span, that's why we have such design. Now I believe it would be just better if we make this statement true again and add `data` to transactions on the Sentry backend.

Reply:
Maybe that's the problem. Having transation inherit from Span.

`data` will definitely not be added on the backend.

Sentry and its SDKs: Transction is event with type=transaction.

We decided to create a class for it and split some fields. Maybe we just don't inherit from Span? 
Composition only in this case.

Reply:
We can think about it but it will probably change the design quite a bit. RIght now, when you get the current span to create a child span, you don't need to worry if it's a transaction or if it's a span - you just get an ISpan that lets you create children.

Reply:
We should discuss this because we'll need to get to a design where spans can be created alone.
Transaction is a container to capture a set of spans (and is a Sentry event) where a Span is not, and one day could be ingested individually (rather in batches) but outside of a Transaction.

Friday we can discuss this if not earlier. I'll see if @mitsuhiko can join us.

Reply:
this is part of our agenda for tomorrow

Reply:
See https://github.com/getsentry/sentry-java/issues/1233#issuecomment-779235496